### PR TITLE
Makes gas actually dissipate

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request
Raises the quantize threshold from 1E-7 to 1E-4.
This makes gas dissipate as expected, and should help with the amount of useless gas floating around the station
at highpop

Adds a garbage_collect() call to the portion of pipeline code where all gasmixes are in one place, this should
clean things up properly.

Changes BREATH_VOLUME from 2 to 1.99. This is imperative
Documents a FUCKING HELLBUG in quantize/breathcode that can lead to breaths just not working sometimes. I'm not
sure how to fix this totally, so I'll document it and pray.
See <https://www.desmos.com/calculator/5icdlnktus>
Adds a unit test to check for this sort of failure.

Addendum for people tweaking this value in the future.
```
Because o2 tank release values/human o2 requirements are very strictly set to the same pressure, small errors
can cause breakage
This comes from QUANTIZE being used in /datum/gas_mixture.remove(), forming a slight sawtooth pattern of the added/removed gas, centered on the actual pressure
Changing BREATH_VOLUME can set us on the lower half of this sawtooth, making humans unable to breath at standard
pressure.
There's no good way I can come up with to hardcode a fix for this. So if you're going to change this variable
graph the functions that describe how it is used/how it interacts with breath code, and pick something on the
upper half of the sawtooth
NOTE: I've made this change with a focus on o2 requirements. Changing this will effect other settings, but most
all of them can be ignored, as none will notice.
```
## Why It's Good For The Game
Freedom from tyranny 
I'm sorry dennok, I didn't believe https://github.com/tgstation/tgstation/pull/50188
## Changelog
:cl:
fix: Gas will actually dissipate now. Screee at me in the issue reports if you see jank around this or breathing (Please don't ask)
/:cl:

### CODE REVIEW

File: code/_onclick/hud/alert.dm [hunk 1]
```diff
@@ -86,6 +86,10 @@
 		client.screen -= alert
 	qdel(alert)
 
+// Proc to check for an alert
+/mob/proc/has_alert(catagory)
+	return !!(alerts[catagory])
```
Comment (Ghilker) [inline code/_onclick/hud/alert.dm:91]: catagory

**Comment by optimumtact** [comment]
start rounding away decimal float points of gas NOW

File: code/_onclick/hud/alert.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/_onclick/hud/alert.dm:91]: we stan spelling

**Comment by maxymax13** [comment]
> Changes BREATH_VOLUME from 2 to 1.99
why? what is this changed for?

**Comment by LemonInTheDark** [comment]
> 
> 
> > Changes BREATH_VOLUME from 2 to 1.99
> 
> can you make it more clear why this is being changed in the pr body for anyone that isn't a nerd

Quantize make sawtooth pattern, describing the pressure breathing 16kpa of o2 actually reads as. Changing quantize constant changes sawtooth frequency. 1E-7 to 1E-4 puts a breath volume of 2 in the negative. 1.99 however is well in the positive. Eternal suffering

File: code/_onclick/hud/alert.dm [hunk 2]
```diff
@@ -86,6 +86,10 @@
 		client.screen -= alert
 	qdel(alert)
 
+// Proc to check for an alert
+/mob/proc/has_alert(category)
+	return !!(alerts[category])
```
Comment (Mothblocks) [inline code/_onclick/hud/alert.dm:91]:
```suggestion
	return !isnull(alerts[category])
```

I'm assuming `alerts[category]` can't be false, in which case no need to get truthiness.

File: code/modules/surgery/organs/lungs.dm [hunk 3]
```diff
@@ -21,7 +21,7 @@
 	food_reagents = list(/datum/reagent/consumable/nutriment = 5, /datum/reagent/medicine/salbutamol = 5)
 
 	//Breath damage
-	//These numbers are roughly equivilant to molar count at room temperature, they scale up and down with temp tho
+	//These numbers are equivilant to pressure taken up by that type of gas in the main mix
```
Comment (Mothblocks) [inline code/modules/surgery/organs/lungs.dm:24]:
```suggestion
	//These numbers are equivalent to pressure taken up by that type of gas in the main mix
```

Comment (Mothblocks) [reply code/modules/surgery/organs/lungs.dm:24]: What main mix though, distro?

File: code/modules/unit_tests/breath.dm [hunk 4]
```diff
@@ -0,0 +1,31 @@
+/datum/unit_test/breath_sanity/Run()
```
Comment (Mothblocks) [inline code/modules/unit_tests/breath.dm:1]: Include this at the top:

```dm
/// Comment explaining your test here
/datum/unit_test/breath_sanity
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/lungs.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/modules/surgery/organs/lungs.dm:24]: I wrote this last night in a stupor, lemme redo it again

File: code/modules/unit_tests/breath.dm (hunk 4 above)
Comment (LemonInTheDark) [reply code/modules/unit_tests/breath.dm:1]: 👍

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**